### PR TITLE
test: add case insensitive

### PR DESCRIPTION
**Pull request by WellingGuzman**
Add some test for `Basic` case insensitivity.

### CODE REVIEW

**Comment by wesleytodd** [comment]
Hey @WellingGuzman,

[The specification](
https://tools.ietf.org/html/rfc2617#section-2) says `Basic`, not `basic` or `BaSic`.  I do not use basic auth very much, so am not an expert in the different implementations out there, but I would think sticking strictly with `Basic` would be best, since it is closest to the spec.

**Comment by dougwilson** [comment]
RFC 2617 has been replaced by https://tools.ietf.org/html/rfc7617 This RFC clarifies that it is supposed to be case-insensitive:

> The scheme name is "Basic".

> Note that both scheme and parameter names are matched case-insensitively.

File: test/basic-auth.js [hunk 1]
```diff
@@ -109,6 +109,29 @@ describe('auth(req)', function () {
...
+    })
+
+    it('should return .name and .pass', function () {
+      var req = request('BASIC Zm9vOmJhcg==')
+      var creds = auth(req)
+      assert.strictEqual(creds.name, 'foo')
+      assert.strictEqual(creds.pass, 'bar')
+    })
+
+    it('should return .name and .pass', function () {
```
Comment (dougwilson) [inline test/basic-auth.js]: When the test suite is run, all these tests appear with exactly the same name & path, making a failure when it is summed up at the bottom had to tell which case failed. Can you either (a) separate these into different `describe` blocks or (b) change the names for the `it` to be different for each within the same `describe` block?

Comment (WellingGuzman) [reply test/basic-auth.js]: Hey @dougwilson, I changed each test case (`it`) appending the word is testing, hopefully this is clearer.

Ref: https://github.com/jshttp/basic-auth/pull/41/commits/fb21123a2dedea7fe2560531e38d3b24ec719f10

**Pull request merged**